src/gpuarray_array.c

```diff
@@ -1008,15 +1009,18 @@ int GpuArray_concatenate(GpuArray *r, const GpuArray **as, size_t n,
 
   res_off = r->offset;
   res_dims = r->dimensions;
+  res_flags = r->flags;
+  r->flags &= ~(GA_C_CONTIGUOUS|GA_F_CONTIGUOUS);
```

Comment:
It is much simpler to fix after the collapsing. As it is faster then then
old back-end, we don't need to speed it up more, but at least add a comment
here that we could check here if the strides are contiguous or not instead
of doing what we do here.

On Thu, May 19, 2016 at 12:23 PM, abergeron <notifications@github.com>
wrote:

> In src/gpuarray_array.c
> <https://github.com/Theano/libgpuarray/pull/178#discussion_r63910102>:
>
> > @@ -1008,15 +1009,18 @@ int GpuArray_concatenate(GpuArray *r, const GpuArray **as, size_t n,
> >
> >    res_off = r->offset;
> >    res_dims = r->dimensions;
> > +  res_flags = r->flags;
> > +  r->flags &= ~(GA_C_CONTIGUOUS|GA_F_CONTIGUOUS);
>
> This is a corner case that doesn't happen anywhere in practice because all
> operations set the flags correctly on the arrays. Here we muck with the
> array properties directly and therefore we have to adjust the flags to
> match. In the interest of code size and speed I err on the safe side and
> just always say that the array is not contiguous because the non-contiguous
> code is safe in the contiguous case and the contiguous case is rare in this
> instance.
>
> I oppose adding code to handle what is technically a violation of the
> interface.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/libgpuarray/pull/178/files/7be54760f35fb3c7b2698d66c532d95528cf762d..04c713b21efa469ae4fa1e356f9b977eb39ceafb#r63910102>
>